fixed delayuntil in kauri's begin

hanish520 opened the pull request:
Fixes #267

hanish520 committed:
fixed delayuntil in kauri begin

meling reviewed (changes requested):
I wonder if it would be possible to get Copilot to prepare some test cases for the Kauri implementation. Perhaps guiding it to use the `clique_test.go` as a basis? Could you give it a go @hanish520?

hanish520 committed:
added tests generated from copilot and fixed comments

meling committed:
refactor: contribution handling to use kauripb.Contribution directly

This removes the container kauri.ContributionRecvEvent which only
wrapped the kauripb.Contribution message; we can simply pass the
protobuf message directly via the event loop since events can be
of 'any' type.

meling committed:
refactor: simplify kauri_test.go

meling committed:
feat: add support for KauriSender interface to MockSender

meling committed:
test: refactored the kauri tests

meling committed:
chore: upgraded go.mod dependencies, except gorums

Gorums contains breaking changes, so we need to do it separately.

meling committed:
chore: regenerated proto files (no real change; just version number)

meling reviewed (approved):
Looks good to me!
